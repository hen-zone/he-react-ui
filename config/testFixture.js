import createTestContext from 'react-cosmos-test/enzyme';

export default function testFixture(fixture, variant) {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component;
  mount();
  test(`<${displayName ||
    name} /> rendered correctly with ${variant} fixture`, () =>
    expect(getWrapper()).toMatchSnapshot());
}
