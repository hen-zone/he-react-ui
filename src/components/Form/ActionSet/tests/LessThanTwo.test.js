import fixture from '../fixtures/LessThanTwo.fixture.js'
import testFixture from '../../../../../config/testFixture'

testFixture(fixture, 'LessThanTwo')
