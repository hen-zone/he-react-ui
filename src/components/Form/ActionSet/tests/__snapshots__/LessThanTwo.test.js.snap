// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnClickOutside(ActionSet) /> rendered correctly with LessThanTwo fixture 1`] = `
<OnClickOutside(ActionSet)
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
>
  <ActionSet
    className=""
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    expanded={false}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
  >
    <div
      className="outer"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="actions"
      >
        <div
          className="action"
          key="0/.0"
          onClick={[Function]}
        >
          <ActionIcon
            className=""
            color="red"
            disabled={false}
            icon="Delete"
            onClick={[Function]}
            title="Delete"
          >
            <button
              className="button red"
              onClick={[Function]}
              title="Delete"
              type="button"
            >
              <Icon
                className="icon"
                name="Delete"
              >
                <Delete
                  className="icon icon"
                  fill="currentColor"
                  height={32}
                  width={32}
                >
                  <svg
                    className="icon icon"
                    fill="currentColor"
                    height={32}
                    viewBox="0 0 24 24"
                    width={32}
                  >
                    <path
                      d="M7.5 5.5V3a1 1 0 0 1 1-1h7a1 1 0 0 1 1 1v2.5H21a1 1 0 0 1 0 2h-1.5v14a1 1 0 0 1-1 1h-13a1 1 0 0 1-1-1v-14H3a1 1 0 1 1 0-2h4.5zm8 2h-9v13h11v-13h-2zM11 11a1 1 0 1 1 2 0v5a1 1 0 0 1-2 0v-5zm-3 0a1 1 0 1 1 2 0v5a1 1 0 0 1-2 0v-5zm6 0a1 1 0 0 1 2 0v5a1 1 0 0 1-2 0v-5zM9.5 5.5h5V4h-5v1.5z"
                    />
                  </svg>
                </Delete>
              </Icon>
            </button>
          </ActionIcon>
        </div>
        <div
          className="action"
          key="1/.1"
          onClick={[Function]}
        >
          <ActionIcon
            className=""
            color="teal"
            disabled={false}
            icon="Edit"
            onClick={[Function]}
            title="Edit"
          >
            <button
              className="button teal"
              onClick={[Function]}
              title="Edit"
              type="button"
            >
              <Icon
                className="icon"
                name="Edit"
              >
                <Edit
                  className="icon icon"
                  fill="currentColor"
                  height={32}
                  width={32}
                >
                  <svg
                    className="icon icon"
                    fill="currentColor"
                    height={32}
                    viewBox="0 0 24 24"
                    width={32}
                  >
                    <path
                      d="M17.5 10.086L19.086 8.5 15.5 4.914 13.914 6.5l3.586 3.586zM16.086 11.5L12.5 7.914l-8 8V19.5h3.586l8-8zM8.5 21.5h-5a1 1 0 0 1-1-1v-5a1 1 0 0 1 .293-.707l12-12a1 1 0 0 1 1.414 0l5 5a1 1 0 0 1 0 1.414l-12 12a1 1 0 0 1-.707.293z"
                    />
                  </svg>
                </Edit>
              </Icon>
            </button>
          </ActionIcon>
        </div>
      </div>
    </div>
  </ActionSet>
</OnClickOutside(ActionSet)>
`;
